### ISSUE TITLE
how does the building query process look like?

### ISSUE THREAD

**Issue by remigium** [slawascichy/hgdb-grafana-datasource#8]
Jak wygląda proces budowania zapytania do Hgdb? 
Wyobrażenie jest takie:
PROJEKCJA
- wybieram nazwę pola z listy pól (-> lista pól z ?)
WARUNKI
- wybieram nazwę pola z listy pól (-> lista pól z ?)
- wybieram operator 
- wybieram wartość / ew. wzorzec wartości
- wybieram spójnik logiczny (opcjonalnie i wtedy powtarzam całość od góry)

Bo ogólnie rozumiem, że chcemy mieć edytor, który pozwoli budować zapytania do Lucene i to dowolne zapytania?

**Comment by remigium** [comment]
czym się różnią dokumenty od spraw?

**Comment by remigium** [comment]
Tutaj coś znalazłem:
https://hgdb.org/confluence/display/MDH3/Zapytania+indeksu+Lucene
czy konieczny jest wybór sprawy do wyszukiwania?

**Comment by slawascichy** [comment]
> czym się różnią dokumenty od spraw?

Przede wszystkim: 
- Sprawa to opis bez fizycznego pliku, 
- Dokument to opis + plik.

Czyli Sprawy to sam opis :).

Z punktu widzenia organizacji przechowywania metadanych w HgDB - niczym. Metadane są rozróżniane flagą na poziomie definicji typu:
```java
	/** Czy jest to typ dokumentu */
	@Column(name = "IS_DOCUMENT_TYPE")
	@Property
	@MappedProperty
	@MappedPropertyEager
	private Boolean isDocumentType;
```
Jednakże na początku był inny cel przechowywania danych o typach dokumentów. Obecnie same dokumenty i ich dane przechowywane są w zewnętrznym repozytorium dokumentów obsługujących CMIS: takich jak Alfresco, FileNet, SharePoint.

Ale przyszłość będzie inna, kwestia może końca roku i dokumenty również będą przechowywane w HgDB.
Wtedy nie będzie różnicy z punktu widzenia organizacji danych (struktury) ale:
1. Dokumenty będą miały swój indeks Lucene
2. Dostęp do dokumentów będzie realizowany przez API CMIS
3. Aby wyszukiwać dane dokumentów będzie trzeba ustawić specjalny parametr w kontekście - na razie w dopracowaniu.

**Comment by slawascichy** [comment]
> czy konieczny jest wybór sprawy do wyszukiwania?

@remigium  nie rozumiem tego pytania

**Comment by slawascichy** [comment]
> Bo ogólnie rozumiem, że chcemy mieć edytor, który pozwoli budować zapytania do Lucene i to dowolne zapytania?

@remigium Tak, to jedno z głównych celów tego projektu. Utworzyłem zadanie #13 to w nim będziemy dyskutować na temat GUI, jak to ma działać.

**Comment by remigium** [comment]
> > czy konieczny jest wybór sprawy do wyszukiwania?
> 
> @remigium nie rozumiem tego pytania

czy przy budowie zapytania obowiązkową składową jest wybór typu spraw (a nie sprawy - wkradł się skrót myślowy)?

**Comment by slawascichy** [comment]
Kryterium typu sprawy w zapytaniach nie jest obowiązkowe. Trzeba mieć tylko świadomość, że w wyniku wyszukiwania otrzymamy sprawy różnych typów i uwzględnić to podczas prezentacji.
